### PR TITLE
Assignment 2 @VeraKorchemnaya @AndrewKorchemniy

### PR DESCRIPTION
**Pull request by AndrewKorchemniy**
https://brave-stone-07b629c1e.3.azurestaticapps.net/

### CODE REVIEW

**Comment by StrangeRanger** [comment]
## Assignment 2

The purpose of this assignment is to solidify your learning of:
- Front end component development
- Using Vuetify's components
- Building a new custom component
- Creating new pages

## Instructions
Note: Custom CSS can only be used where specified.

1. Create a custom component to show available words ❌
- ***REVIEW NOTE: It looks like you have the UI and foundation for this started, but it's currently non-functional.***
- Create/extend a word list component with a method called validWords that returns an array of valid words based on current guesses ❌
- Create unit tests for the validWords method ❌
- Display the number of valid words based on the entered letters ❌
- When the number is clicked, show a dialog with all the available words with scrolling if there are too many to show on the screen. ❌
- When a word is clicked, the word should be entered as the guess but not submitted ❌
- Add a hover over for the word so it is obvious that it can be clicked. (No CSS) ✅
  - ***REVIEW NOTE: It's very faint and hard to tell that something changes when hovering over the button.*** 

3. Convert the sidebar to an App Bar ✅
- Name of the app on the left with an icon ✅
- Choose an icon for the app (from Material Design or somewhere on the web) ✅
- Clicking the name and icon of the app take you to the home page ✅
- Hamburger menu on the right ✅
- Menu option for an About page ✅
  - Create an About page that has a few sentences about this class project and how great Meg is ✅
  - Make sure this has a back button ✅

4. Add a settings dialog ✅
- Add a settings selection to the App Bar's menu with a gear icon. This should open this dialog regardless of where you are on the site ✅
- Option for dark and light mode ✅
- Develop two additional color schemes (with creative names) and allow the user to change to them. Schemes should look good in both light and dark mode ✅
- The above two items must be implemented with built in Veutify features ✅

5. Style the main game page ✅
- Make it look like [wordle.com](https://www.nytimes.com/games/wordle/index.html) or the Wordle mobile app ✅
- Add gradient to guesses blocks and keys (with CSS) ✅
- Add drop shadow to keys (No CSS) ✅
- Make the app responsive (No CSS) ✅

6. Include a link to your Azure App in your PR comments.

~~## Extra Credit~~

- ~~Add an appropriate sound when the UI letter buttons are clicked (5 points)~~
- ~~Add creative styling following good UI/UX practices~~
  - ~~Look at products by Google or some of the vuetify components for good UX/UI ideas~~
  - ~~https://m2.material.io/design/guidelines-overview~~

## Other

Overall, you did a fantastic job. I like the choices you made for how both the navigation links and the themes are displayed. Everything looks very neat and cleanly designed. One thing I'd recommend is settings a maximum width for the main content. Specifically on the About page, the width expands the whole webpage, but generally, most websites have a maximum width, for example, 1280px.

**Comment by StrangeRanger** [comment]
@AndrewKorchemniy @VeraKorchemnaya Please note I think I made a mistake in terms of whether the app is responsive; I'm not 100% whether it is or it is not. Also, I do plan later today to go through the code more thoroughly and provide possible suggestions. Just didn't have time when I initially wrote everything above.

**Review by StrangeRanger — COMMENTED** [review]
I feel like there is possibly more I could make notes about, but I'm finding it a bit difficult find them. Sorry I couldn't really provide any better feedback.

File: wordle-mind-bender/src/components/AvailableWords.vue [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+  },
+  {
+    title: 'Item #3',
+    value: 3
+  },
+  {
+    title: 'Item #4',
+    value: 4
+  }
+]
```
[the remark this answers is not in the document]
Comment (VeraKorchemnaya) [reply wordle-mind-bender/src/components/AvailableWords.vue]: Yes, thank you! Me and Andrew are working to get the validWords() function working so we can return some real data. Also thank you for pointing out the width on the About page, I don't know how we missed that! You mentioned responsiveness earlier, we also need to figure that out especially for the keyboard. I'm also glad you mentioned the hover on the available words list, I agree, it is very faint, I will go ahead and fix that.
Over all very helpful feedback, thank you!

File: wordle-web/src/components/ThemeSelector.vue [hunk 2]
```diff
@@ -0,0 +1,64 @@
+<template>
+  <v-dialog transition="dialog-bottom-transition" width="30rem">
+    <template v-slot:activator="{ props }">
+      <v-btn v-bind="props" class="ml-1" v-on:keydown.enter.prevent
+        ><v-icon icon="mdi-cog" size="x-large"></v-icon
```
Comment (mmwoodfo) [inline wordle-web/src/components/ThemeSelector.vue:5]:
```suggestion
        > <v-icon icon="mdi-cog" size="x-large"> </v-icon
```
If you put spaces between these, your linter will format it a bit better.

File: wordle-web/src/App.vue [hunk 3]
```diff
@@ -1,28 +1,45 @@
...
-import { useTheme } from 'vuetify/lib/framework.mjs'
+import { RouterView } from 'vue-router'
+import ThemeSelector from './components/ThemeSelector.vue'
+</script>
 
-const theme = useTheme()
+<template>
+  <v-container fluid class="pa-0">
+    <v-row class="ma-0">
+      <v-col class="pa-0">
```
Comment (mmwoodfo) [inline wordle-web/src/App.vue:9]: I'd suggest not using v-row and v-col unless you are going to have multiple v-col elements for a grid system.

**Review by mmwoodfo — COMMENTED** [review]
Looks like there's a bit of a bug with your suggested word list, guessing close to the word produces no suggested words, otherwise it seems to work great! Good job, this was a difficult problem. 
image

**Pull request closed without merging**